File: uniffi/src/message/groups/trans_receipt_quadruples.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::{Prefixer, Seqner, Saider, Siger};
pub use cesrox::message::groups::{
    TransReceiptQuadruples,
    TransReceiptQuadruple,
};

pub fn trans_receipt_quadruple_create(prefixer: Prefixer, seqner: Seqner, saider: Saider, siger: Siger) -> TransReceiptQuadruple {
    TransReceiptQuadruple::new(prefixer, seqner, saider, siger)
}

pub fn trans_receipt_quadruples_create(value: Vec<TransReceiptQuadruple>) -> TransReceiptQuadruples {
    TransReceiptQuadruples::new(value)
}

pub fn trans_receipt_quadruples_qb64(trans_receipt_quadruples: &TransReceiptQuadruples) -> ParsideResult<String> {
    trans_receipt_quadruples.qb64()
}

pub fn trans_receipt_quadruples_qb64b(trans_receipt_quadruples: &TransReceiptQuadruples) -> ParsideResult<Vec<u8>> {
    trans_receipt_quadruples.qb64b()
}

pub fn trans_receipt_quadruples_qb2(trans_receipt_quadruples: &TransReceiptQuadruples) -> ParsideResult<Vec<u8>> {
    trans_receipt_quadruples.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/TransReceiptQuadruples.kt
package cesrox.groups

import cesrox.primitives.Prefixer
import cesrox.primitives.Seqner
import cesrox.primitives.Saider
import cesrox.primitives.Siger
import uniffi.cesrox.*
import uniffi.cesrox.TransReceiptQuadruple as TransReceiptQuadrupleNative
import uniffi.cesrox.TransReceiptQuadruples as TransReceiptQuadruplesNative

class TransReceiptQuadruple {
    val value: TransReceiptQuadrupleNative

    constructor(prefixer: Prefixer, seqner: Seqner, saider: Saider, siger: Siger) {
        value = transReceiptQuadrupleCreate(prefixer.value, seqner.value, saider.value, siger.value)
    }

    constructor(nativeValue: TransReceiptQuadrupleNative) {
        value = nativeValue
    }

    val prefixer get() = Prefixer(value.prefixer)

    val seqner get() = Seqner(value.seqner)

    val saider get() = Saider(value.saider)

    val siger get() = Siger(value.siger)
}

class TransReceiptQuadruples {
    val value: TransReceiptQuadruplesNative

    constructor(couples: List<TransReceiptQuadruple>) {
        value = transReceiptQuadruplesCreate(couples.map { it.value })
    }

    constructor(couples: TransReceiptQuadruplesNative) {
        value = couples
    }

    fun get(index: Int): TransReceiptQuadruple {
        return TransReceiptQuadruple(value.value[index])
    }

    fun qb2(): List<UByte> {
        return transReceiptQuadruplesQb2(value)
    }
    fun qb64b(): List<UByte> {
        return transReceiptQuadruplesQb64b(value)
    }
    fun qb64(): String {
        return transReceiptQuadruplesQb64(value)
    }
}